modules/google_service_account/iam-memberships.tf

```diff
@@ -0,0 +1,84 @@
+locals {
+  cross_project_iam_memberships = flatten([
+    for project_id, memberships in var.cross_project_iam_memberships : [
+      for membership in memberships : {
+        project_id = project_id
+        role       = membership.role
+        conditions = membership.conditions
+      }
+    ]
+  ])
+
+  folder_iam_memberships = flatten([
+    for folder_id, memberships in var.folder_iam_memberships : [
+      for membership in memberships : {
+        folder_id  = folder_id
+        role       = membership.role
+        conditions = membership.conditions
+      }
+    ]
+  ])
+}
+
+resource "google_project_iam_member" "project_iam_memberships" {
```

Comment:
@MXfive is there any advantage in separating these two resources - the in-project vs cross-project bindings/memberships?

Reply:
Not really, just felt more intuitive and obvious when reading the code and using it.  Not got a strong opinion either way.